Fade out content that overflow out of the top and bottom edges of a dialog's content

Kiarokh opened the pull request:
fix https://github.com/Lundalogik/lime-elements/issues/1998

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

Kiarokh commented on src/components/dialog/dialog.scss:
```diff
@@ -16,7 +17,7 @@
  * @prop --dialog-max-height: Max height of the dialog.
  * @prop --dialog-border-radius: Border radius of the dialog corners
- * @prop --dialog-padding-top-bottom: Padding on top and bottom of dialog content. Defaults to `1.5rem`.
+ * @prop --dialog-padding-top-bottom: Padding on top and bottom of dialog content. Defaults to `1.5rem`. Note that if you use this variable and set it to numbers smaller than 1rem, you will loose the edge fade effects when the content is scrollable and has overflowed out of the dialog's content area. This should be however fine! Because you must then be handling the overflow internally in the com ...
```
Maybe I should also explain how these affect the fade-out effect of the edges, generally.

LucyChyzhova replied:
I think this is a good explanation. Maybe we could add the preferred padding ranges.

Kiarokh replied:
Maybe. The more padding, the larger the fade-out effect will be.

Kiarokh commented on src/components/dialog/dialog.scss:
```diff
@@ -16,7 +17,7 @@
  * @prop --dialog-max-height: Max height of the dialog.
  * @prop --dialog-border-radius: Border radius of the dialog corners
- * @prop --dialog-padding-top-bottom: Padding on top and bottom of dialog content. Defaults to `1.5rem`.
+ * @prop --dialog-padding-top-bottom: Padding on top and bottom of dialog content. Affects the height of fade-out effects on top and bottom edges when the content is scrollable and has overflowed out of the content area. Defaults to `1.5rem`. Note that if you use this variable and set it to numbers smaller than 1rem, you will loose the fade-out effects on the edges. If you have set these paddings ...
```
it might be hard to know what these docs are trying to say. but this is the background story of adding those variables: https://github.com/Lundalogik/lime-elements/issues/1953 some appers user the whole dialog surface to render their "fullscreen apps", and they don't want our paddings. They just set it to zero: https://github.com/search?q=org%3ALundalogik%20--dialog-padding-top-bottom&type=code

adrianschmidt commented:
This version has its own problems:

https://github.com/Lundalogik/lime-elements/assets/154725/b3f8198a-5b4d-44e4-a866-ed1e243fccc8

But if we think this version is less problematic than the old one, I'm happy to merge it 👍

Kiarokh commented:
You're right. The perfect solution would be having a reusable script that adds some sort of gradient shadow  on the edge on each side/edge of the scroll area when the content has scrolled out (and also manipulates the gradient masks using the CSS variables that I added in the mixin). However, I can't write such a script myself. If you wanna join me for a session, I can explain what I mean.

However, as you said this is less problematic than what we had, and more globally usable. I added a new commit where we use the same mixin in the recently added breadcrumbs component, to show case the reusability of this mixin better.

lime-opensource commented:
:tada: This PR is included in version 37.1.0-next.30 :tada:

The release is available on:
- [npm package (@next dist-tag)](https://www.npmjs.com/package/@limetech/lime-elements/v/37.1.0-next.30)
- [GitHub release](https://github.com/Lundalogik/lime-elements/releases/tag/v37.1.0-next.30)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

adrianschmidt commented:
> The perfect solution would be having a reusable script that adds some sort of gradient shadow on the edge on each side/edge of the scroll area when the content has scrolled out (and also manipulates the gradient masks using the CSS variables that I added in the mixin). However, I can't write such a script myself. If you wanna join me for a session, I can explain what I mean.

I'm think I know exactly what you mean. I've been thinking more or less the same myself. Let's do it some day! Next hackday maybe? This is an improvement in the meantime 👍

lime-opensource commented:
:tada: This PR is included in version 37.1.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@limetech/lime-elements/v/37.1.0)
- [GitHub release](https://github.com/Lundalogik/lime-elements/releases/tag/v37.1.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

lime-opensource commented:
:tada: This PR is included in version 37.1.0 :tada:

The release is available on:
- [npm package (@release-37.1.x dist-tag)](https://www.npmjs.com/package/@limetech/lime-elements/v/37.1.0)
- [GitHub release](https://github.com/Lundalogik/lime-elements/releases/tag/v37.1.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
